[feat][Angel] front-end of view cases

royalsprites opened the pull request:
**View Cases (frontend)**

![image](https://github.com/user-attachments/assets/9dcc744b-404f-4452-abee-b539f9c04a10)

![image](https://github.com/user-attachments/assets/aa2194d4-6b12-46d4-85d8-d96856790c0d)

![image](https://github.com/user-attachments/assets/10fd1ba4-fc51-4f9a-afc3-49c850202ca0)

![image](https://github.com/user-attachments/assets/d3f0b09f-efd0-434b-9b09-813afb836fc0)

![image](https://github.com/user-attachments/assets/387e97b9-2d18-408a-8cb6-8164135b7fa3)

royalsprites committed:
[feat][Angel] front-end of view cases, changes in BaseFormInput and BaseFormSelect (for width input)

royalsprites committed:
[feat][Angel] front-end of view cases, changes in BaseFormInput and BaseFormSelect (for width input)

lssndrrr commented on src/app/dashboard/case/page.tsx:
```diff
@@ -1,11 +1,349 @@
+  );
+};
+
+const CaseCard: React.FC<{ caseItem: Case }> = ({ caseItem }) => (
+  <article className="mt-8 z-0">
+    <div className="relative ml-[580px] flex justify-center items-center px-1.5 py-1 bg-white border border-gray-300 rounded-lg shadow-sm w-fit z-0">
+      <span className={`mr-2 flex shrink-0 w-2 h-2 ${getCategoryColor(caseItem.category.name)} rounded-full`} aria-hidden="true"></span>
+      <span className="mr-2 text-[9.8px] font-bold">{caseItem.category.name}</span>
+    </div>
+    <div className="flex z-0 gap-5 justify-between items-start px-9 py-7 mt-[-10px] w-full text-black bg-white rounded-3xl shadow-lg border border-black-100 border-opacity-90">
+      <img src={caseItem.avatar} alt="" className="w-[70px] rounded-full" />
+      <div className="flex flex-col text-xs">
+        <h2 className="text-xl font-bold">{caseItem.title}</h2>
+        <div className="flex gap-1 mt-3 font-light">
+          <img
+            src="https://cdn.builder.io/api/v1/image/assets/TEMP/f62326b4ac842495a4827f224d60986da492d5c8?placeholderIfAbsent=true&apiKey=b97adb845ad745fdabf283f95e3c166e"
+            alt=""
+            className="w-2"
+            aria-hidden="true"
+          />
+          <p>
+            Latest update from <strong className="font-bold">{caseItem.lastUpdate.user}</strong> {caseItem.lastUpdate.time}
+          </p>
+        </div>
+        <p className="mt-4 font-[275]">{caseItem.description}</p>
+      </div>
+      <div className="flex gap-2 px-2.5 py-2.5 text-base font-bold bg-white rounded-xl shadow-[0px_0px_4px_rgba(0,0,0,0.25)] hover:bg-blue-700">
+        <StatusIndicator isOpen={caseItem.status.isOpen} />
+        <span className="pl-1 pr-4  text-base font-extrabold">{caseItem.status.isOpen ? "Open" : "Close"}</span>
+      </div>
+    </div>
+  </article>
+);
```
Transfer to a separate component file

lssndrrr commented on src/app/dashboard/case/page.tsx:
```diff
@@ -1,11 +1,349 @@
+import { useRouter } from "next/navigation";
 
-const Page: React.FC = () => {
+interface Case {
+  id: string;
+  title: string;
+  category: {
+    name: string;
+    color: string;
+  };
+  status: {
+    isOpen: boolean;
+    color: string;
+  };
+  avatar: string;
+  lastUpdate: {
+    user: string;
+    time: string;
+  };
+  description: string;
+}
```
Move to `interface` folder

Lanieza committed:
[feat][lanie] create case type interface

royalsprites committed:
[feat][Angel] created separate component for CaseCard and some minor changes

royalsprites committed:
[feat][Angel] created separate component for CaseCard and some minor changes

royalsprites commented:
Minor changes:

(blue text and check when pressed)
![image](https://github.com/user-attachments/assets/98082e4c-a109-4a7c-9ce5-98ddf824d0c0)

royalsprites committed:
[feat][Angel] created a separate component file for Header under Profile and added a few functions in case page.tsx

royalsprites committed:
[feat][Angel] created a separate component file for Header under Profile and added a few functions in case page.tsx

royalsprites commented:
Current changes:

![image](https://github.com/user-attachments/assets/47501d73-04bf-4117-a30b-cdf1a78bab6c)

![image](https://github.com/user-attachments/assets/3a2c1062-2752-462c-9686-9b3efb487c23)

![image](https://github.com/user-attachments/assets/25469446-5abc-49c7-bb21-2077d6f690cf)

![image](https://github.com/user-attachments/assets/844ffc88-b790-414a-b873-ebd81dfcb732)

![image](https://github.com/user-attachments/assets/5584ff26-3a76-4e2f-bdc1-0b6bfb008853)

royalsprites committed:
[feat][Angel] minor changes in notifications

Lanieza committed:
[feat][lanie] Created view-cases API and performed a large code refactor in the backend

Lanieza committed:
[feat][lanie] Created view-cases API and performed a large code refactor in the backend

Lanieza committed:
[feat][lanie] made changes to header.tsx, reflected it to the new field from case types interface

Lanieza committed:
[feat][lanie] fixed the case status position

Lanieza committed:
[feat][lanie] fixed the case status position

lssndrrr committed:
[refactor][az] modularizes browse case component

lssndrrr committed:
[refactor][az] modularizes browse case component

ndb3llo committed:
Merge branch 'feat/view-cases' of https://github.com/ctrlsolutions/laymans-law into feat/view-cases

ndb3llo committed:
[feat][Nico] Creates AcceptCaseModal first version. Need to make Accept and Decline buttons component pa]

ndb3llo committed:
[feat][Nico] Creates AcceptCaseModal first version. Need to make Accept and Decline buttons component pa]

cajober committed:
[feat][caleb] initial commit

cajober committed:
[feat][caleb] edits (account)/layout.tsx

lssndrrr committed:
[merge][az] pull request #106 from ctrlsolutions/feat/routing

[feat][caleb] adds placeholder files for wiki, ofw support, forum in the (account) route

Lanieza committed:
Merge branch 'feat/view-cases' of https://github.com/ctrlsolutions/laymans-law into feat/view-cases

Lanieza committed:
[feat][lanie] removed the dashboard/case/ page and fixed the error in browse case component

Lanieza committed:
[feat][lanie] removed the dashboard/case/ page and fixed the error in browse case component

Lanieza committed:
[feat][lanie] sidebar component and header component

Lanieza committed:
[feat][lanie] added casetypes.ts

Lanieza committed:
[feat][lanie] added casetypes.ts

Lanieza committed:
[feat][lanie] update case types

Lanieza committed:
[feat][lanie] added case filter button

Lanieza committed:
[feat][lanie] case types

Lanieza committed:
[refactor][lanie] centralize shared interfaces in CaseTypes.ts

Lanieza committed:
[chore][lanie] removed comments

lssndrrr committed:
[merge][az] pull request #107 from ctrlsolutions/feat/sidebar

Feat/sidebar

Lanieza committed:
[feat][lanie] updated casecard and sidebar color map to sync

Lanieza committed:
Merge branch 'feat/view-cases' of https://github.com/ctrlsolutions/laymans-law into feat/view-cases

Lanieza committed:
[chore][lanie] removed inner shadow in browse case components
